qiskit/providers/ibmq/job/ibmqjob.py

```diff
@@ -23,9 +23,8 @@
 
 from marshmallow import ValidationError
 
-from qiskit.providers.basejob import BaseJob
-from qiskit.providers.basebackend import BaseBackend
-from qiskit.providers.exceptions import JobTimeoutError
+from qiskit.providers import (BaseJob, # type: ignore[attr-defined]
+                              JobTimeoutError, BaseBackend)
```

Comment:
Note I reverted this change from `master` during the conflict solving at the latest merge, as technically the `qiskit.providers` API is the one that is stable - we don't have that guarantee for `qiskit.providers.xxx.yyy` (at the expense of the mypy ignore).